### DISCUSSION TITLE
Help deploying milvus2.0.0rc7 cluster on openshift

### DISCUSSION THREAD

**Discussion by DrBamba** [milvus-io/milvus#12193] [Q&A and General discussion] [answered]
I am trying to deploy a cluster version on openshift.
To do so I created deployments for every service of the cluster in order to be able to scale them separately. (As far as I understand all I need/can scale is the node services).

When the proxy starts it says the correct RootCoord address that I set in the configs(due to them not being on the same deployment they are not both on the same localhost) 
But immediately afterwards it starts going in a loop of RootCoordClient try reconnect and says TRANSIENT_FAILURE 
It also shows that the address is now localhost:53100 instead of the correct address of the rootcoord (it says the address at the line that reads "RootCoordClient try reconnect")

Could not find an answer anywhere would really appreciate some help

milvus-proxy logs:
[DEBUG] [basetable.go:56] ["config directory"] [configDir=/milvus/configs/]
[DEBUG] [proxy.go:96] [Proxy] [State=Abnormal]
[DEBUG] [param_table.go:140] [initServerMaxSendSize] [proxy.grpc.serverMaxSendSize=2147483647]
[DEBUG] [param_table.go:163] [initServerMaxRecvSize] [proxy.grpc.serverMaxRecvSize=2147483647]
[DEBUG] [basetable.go:56] ["config directory"] [configDir=/milvus/configs/]
[DEBUG] [service.go:137] ["init params done ..."]
[DEBUG] [service.go:150] [proxy] ["proxy host"=10.112.21.149]
[DEBUG] [service.go:151] [proxy] ["proxy post"=19530]
[DEBUG] [service.go:152] [proxy] ["proxy address"=10.112.21.149:19530]
[DEBUG] [session_util.go:71] ["Session try to connect to etcd"]
[DEBUG] [session_util.go:90] ["Session connect to etcd success"]
[DEBUG] [session_util.go:119] ["Session checkIDExist Begin"]
[DEBUG] [session_util.go:126] ["Session checkIDExist End"]
[DEBUG] [session_util.go:131] ["Session try to get serverID"]
[DEBUG] [session_util.go:162] ["Session get serverID success"]
[DEBUG] [session_util.go:182] ["Session Register Begin"]
[DEBUG] [session_util.go:221] ["Session Register End"] [ServerID=241]
[INFO] [roles.go:479] ["Set log file to "] [path=]
[DEBUG] [service.go:84] [proxy] ["network port"=19530]
[DEBUG] [service.go:164] ["create grpc server ..."]
[DEBUG] [service.go:170] [Proxy] ["RootCoord address"=milvus-rootcoord.project.svc:53100]
[DEBUG] [session_util.go:71] ["Session try to connect to etcd"]
[INFO] [grpclog.go:37] ["[core]parsed scheme: \"etcd-endpoints\""]
[INFO] [grpclog.go:37] ["[core]ccResolverWrapper: sending update to cc: {[{milvus-etcd.project.svc:2379 milvus-etcd.project.svc <nil> 0 <nil>}] 0xc0000220a0 <nil>}"]
[INFO] [grpclog.go:37] ["[core]ClientConn swirching balancer to \"round_robin\""]
[INFO] [grpclog.go:37] ["[core]Channel switches to new LB policy \"round_robin\""]
[INFO] [grpclog.go:37] ["[core]parsed scheme: \"etcd-endpoints\""]
[INFO] [grpclog.go:37] ["[balancer]base.baseBalancer: got new new ClientConn state: {{milvus-etcd.project.svc:2379 milvus-etcd.project.svc <nil> 0 <nil>}] 0xc0000220a0 <nil>} <nil>}"]
[INFO] [grpclog.go:37] ["[core]Subchannel Connectivity change to CONNECTING"]
[INFO] [grpclog.go:37] ["[core]Subchannel picks a new address \"milvus-etcd.project.svc:2379\" to connect"]
[INFO] [grpclog.go:37] ["[balancer]base.baseBalancer: handle SubConn state change: 0xc00051b100, CONNECTING"]
[INFO] [grpclog.go:37] ["[core]Channel Connectivity change to CONNECTING"]
[INFO] [grpclog.go:37] ["[core]Subchannel Connectivity change to READY"]
[INFO] [grpclog.go:37] ["[balancer]base.baseBalancer: handle SubConn state change: 0xc00051b100, READY"]
[INFO] [grpclog.go:37] ["[roundrobin]roundrobinPicker: newPicker called with info: {map[0xc00051b100:{{milvus-etcd.project.svc:2379 milvus-etcd.project.svc <nil> 0 <nil>}}]}"]
[INFO] [grpclog.go:37] ["[core]Channel Connectivity change to READY"]
[DEBUG] [session_util.go:90] ["Session connect to etcd success"]
[DEBUG] [basetable.go:56] ["config directory"] [configDir=/milvus/configs/]
[DEBUG] [param_table.go:66] [initClientMaxSendSize] [rootCoord.grpc.clientMaxSendSize=104857600]
[DEBUG] [param_table.go:89] [initClientMaxRecvSize] [rootCoord.grpc.clientMaxRecvSize=104857600]
[DEBUG] [session_util.go:269] ["SessionUtil GetSessions"] [prefix=RootCoord] [resp="{\"header\":{\"cluster_id\":14841639068965178418,\"member_id\":10276657743932975437,\"revision\":83455,\"raft_term\":2},\"kvs\":[{\"key\":\"YnktZGV2L21ldGEvc2Vzc2lvbi9Sb290Q29vcmQ=\",\"create_revision\":19278,\"mod_revision\":19278,\"version\":1,\"value\":\"eyJTZXJ2ZXJJRCI6NzAsIlNlcnZlck5hbWUiOiJSb290Q29vcmQiLCJBZGRyZXNzIjoibG9jYWxob3N0OjUzMTAwIiwiRXhjbHVzaXZlIJp0cnVlfQ==\",lease\":7587858691350714253}],\"count\":1}"]
[DEBUG] [client.go:72] ["RootCoordClient GetSessions success"]
[DEBUG] [client.go:120] ["RootCoordClient try reconnect "] [address=localhost:53100]
[INFO] [grpclog.go:37] ["[core]parsed scheme: \"\""]
[INFO] [grpclog.go:37] ["[core]scheme: \"\" registered, fallback to default scheme"]
[INFO] [grpclog.go:37] ["[core]ccResolverWrapper: sending update to cc: {[{localhost:53100 <nil> 0 <nil>}] <nil> <nil>}"]
[INFO] [grpclog.go:37] ["[core]ClientConn swirching balancer to \"pick first\""]
[INFO] [grpclog.go:37] ["[core]Channel switches to new LB policy \"pick first\""]
[INFO] [grpclog.go:37] ["[core]Subchannel Connectivity change to CONNECTING"]
[INFO] [grpclog.go:37] ["[core]Subchannel picks a new address \"localhost:53100\" to connect"]
[INFO] [grpclog.go:37] ["[core]pickfirstBalancer: UpdateSubConnState: 0xc0002f8170, {CONNECTING <nil>}"]
[INFO] [grpclog.go:37] ["[core]Channel Connectivity change to CONNECTING"]
[INFO] [grpclog.go:37] ["[core]grpc: addrConn.createTransport failed to connect to {localhost:53100 localhost:53100 <nil> 0 <nil>}. Err: connection error: desc: \"transport: Error while dialing dial tcp [::1]:53100: connect: connection refused\"}"]
[INFO] [grpclog.go:37] ["[core]Subchannel Connectivity change to TRANSIENT_FAILURE"]
[INFO] [grpclog.go:37] ["[core]pickfirstBalancer: UpdateSubConnState: 0xc0002f8170, {TRANSIENT_FAILURE connection error: desc = \"transport: Error while dialing dial tcp [::1]:53100: connect: connection refused\"}"]
[INFO] [grpclog.go:37] ["[core]Channel Connectivity change to TRANSIENT_FAILURE"]
[INFO] [grpclog.go:37] ["[core]Subchannel Connectivity change to CONNECTING"]
[INFO] [grpclog.go:37] ["[core]Subchannel picks a new address \"localhost:53100\" to connect"]
[INFO] [grpclog.go:37] ["[core]pickfirstBalancer: UpdateSubConnState: 0xc0002f8170, {CONNECTING <nil>}"]
[INFO] [grpclog.go:37] ["[core]Channel Connectivity change to CONNECTING"]
[INFO] [grpclog.go:37] ["[core]grpc: addrConn.createTransport failed to connect to {localhost:53100 localhost:53100 <nil> 0 <nil>}. Err: connection error: desc: \"transport: Error while dialing dial tcp [::1]:53100: connect: connection refused\"}"]
[INFO] [grpclog.go:37] ["[core]Subchannel Connectivity change to TRANSIENT_FAILURE"]
[INFO] [grpclog.go:37] ["[core]pickfirstBalancer: UpdateSubConnState: 0xc0002f8170, {TRANSIENT_FAILURE connection error: desc = \"transport: Error while dialing dial tcp [::1]:53100: connect: connection refused\"}"]
[INFO] [grpclog.go:37] ["[core]Channel Connectivity change to TRANSIENT_FAILURE"]
[INFO] [grpclog.go:37] ["[core]Subchannel Connectivity change to CONNECTING"]
[INFO] [grpclog.go:37] ["[core]Subchannel picks a new address \"localhost:53100\" to connect"]
[INFO] [grpclog.go:37] ["[core]pickfirstBalancer: UpdateSubConnState: 0xc0002f8170, {CONNECTING <nil>}"]
[INFO] [grpclog.go:37] ["[core]Channel Connectivity change to CONNECTING"]
[INFO] [grpclog.go:37] ["[core]grpc: addrConn.createTransport failed to connect to {localhost:53100 localhost:53100 <nil> 0 <nil>}. Err: connection error: desc: \"transport: Error while dialing dial tcp [::1]:53100: connect: connection refused\"}"]
[INFO] [grpclog.go:37] ["[core]Subchannel Connectivity change to TRANSIENT_FAILURE"]
[INFO] [grpclog.go:37] ["[core]pickfirstBalancer: UpdateSubConnState: 0xc0002f8170, {TRANSIENT_FAILURE connection error: desc = \"transport: Error while dialing dial tcp [::1]:53100: connect: connection refused\"}"]
[INFO] [grpclog.go:37] ["[core]Channel Connectivity change to TRANSIENT_FAILURE"]
[INFO] [grpclog.go:37] ["[core]Subchannel Connectivity change to CONNECTING"]
[INFO] [grpclog.go:37] ["[core]Subchannel picks a new address \"localhost:53100\" to connect"]
[INFO] [grpclog.go:37] ["[core]pickfirstBalancer: UpdateSubConnState: 0xc0002f8170, {CONNECTING <nil>}"]
[INFO] [grpclog.go:37] ["[core]Channel Connectivity change to CONNECTING"]
[INFO] [grpclog.go:37] ["[core]grpc: addrConn.createTransport failed to connect to {localhost:53100 localhost:53100 <nil> 0 <nil>}. Err: connection error: desc: \"transport: Error while dialing dial tcp [::1]:53100: connect: connection refused\"}"]
[INFO] [grpclog.go:37] ["[core]Subchannel Connectivity change to TRANSIENT_FAILURE"]
[INFO] [grpclog.go:37] ["[core]pickfirstBalancer: UpdateSubConnState: 0xc0002f8170, {TRANSIENT_FAILURE connection error: desc = \"transport: Error while dialing dial tcp [::1]:53100: connect: connection refused\"}"]
[INFO] [grpclog.go:37] ["[core]Channel Connectivity change to TRANSIENT_FAILURE"]

EDIT: added milvus-proxy logs

**Comment by zwd1208** [comment]
I have not test deployment on OpenShift, but I think it should be compatible with k8s.  And I can't confirm what the problem is from your description, can you give more configuration and pod logs ?

**Reply by DrBamba** [reply]
Hey I added the logs, the configurations are default except I changed the addresses to the deployments addresses such as milvus-roorcoord.project.svc

**Comment by DrBamba** [comment, marked as answer]
Ok it seems that the cause of the issue was that I didn't configure all the services to use the configuration file and so when they boot up they told the etcd a different address for rootcoord(localhost) and so when the proxy reached out to check the rootcoord address it was incorrect.
I changed it so all the services use the configuration file and now it works.
Does the description of the error make sense?
